pkg/templates/promrules.go

```diff
@@ -130,3 +145,61 @@ func (a *PrometheusRuleTemplateManager) CreateFromIngress(ingress *extensionsv1b
 
 	return collectPrometheusRules(prometheusRules), nil
 }
+
+// CreateFromDeployment
+// - Creates all the promRules for a given Deployment
+func (a *PrometheusRuleTemplateManager) CreateFromDeployment(deployment *apps.Deployment, depNamespacePrometheus string) ([]*monitoringv1.PrometheusRule, error) {
+	deploymentIdentifier := fmt.Sprintf("%s.%s", deployment.Namespace, deployment.Name)
+
+	params := &templateParameterDeployment{
+		Deployment:   deployment,
+		Identifier:   deploymentIdentifier,
+		Namespace:    deployment.Namespace,
+		Name:         deployment.Name,
+		Selector:     deployment.GetLabels()["app"],
+		NSPrometheus: depNamespacePrometheus,
+	}
+
+	prometheusRules := map[string]*monitoringv1.PrometheusRule{}
+	annotations := params.Deployment.GetAnnotations()
+
+	for k, v := range annotations {
+		if !strings.HasPrefix(k, heimPrefix) {
+			continue
+		}
+
+		templateName := strings.TrimLeft(k, fmt.Sprintf("%s/", heimPrefix))
+		log.Printf("\n *** templateName is: %s", templateName)
+		template, ok := a.templates[templateName]
+		if !ok {
+			log.Warnf("[deployment][%s] no template for \"%s\"", deploymentIdentifier, templateName)
+			continue
+		}
+
+		params.Threshold = v
+		var result bytes.Buffer
+		if err := template.Execute(&result, params); err != nil {
+			log.Warnf("[deployment][%s] error executing template : %s", deploymentIdentifier, err)
```

Comment:
Since we're using `logrus` why don't we use our fields instead of the `[deployment][%s]` string.

```
 log.WithFields(log.Fields{
    "deployment": deploymentIdentifier,
  }).Warnf("error executing template : %s", err)
```

This should apply anywhere inside of this context, so we could also start the function with the following to include the deployment name on every log call:
```
logger := log.WithFields(log.Fields{"deployment": deploymentIdentifier})
```

and then later just use:
```
logger.Warnf("Something happened! %s", err)
```